src/mongoose_api_common.erl

```diff
@@ -381,3 +382,20 @@ is_known_auth_method(_) -> false.
 
 make_unauthorized_response(Req, State) ->
     {{false, <<"Basic realm=\"mongooseim\"">>}, Req, State}.
+
+-spec check_password(jid:jid() | error, binary()) -> {true, mongoose_credentials:t()} | false.
+check_password(error, _) ->
+    false;
+check_password(JID, Password) ->
+    {LUser, LServer} = jid:to_lus(JID),
+    case mongoose_domain_api:get_domain_host_type(LServer) of
+        {ok, HostType} ->
+            Creds0 = mongoose_credentials:new(LServer, HostType),
+            Creds1 = mongoose_credentials:set(Creds0, username, LUser),
+            Creds2 = mongoose_credentials:set(Creds1, password, Password),
+            case ejabberd_auth:authorize(Creds2) of
+                {ok, Creds} -> {true, Creds};
+                _ -> false
```

Comment:
can we cover auth failure case here? but probably not needed, because it is just moved the old code